lib/msal-core/src/ServerRequestParameters.ts

```diff
@@ -355,6 +351,35 @@ export class ServerRequestParameters {
         return paramsString;
     }
 
+    /**
+     * @hidden
+     *
+     * Validate  scopes/state/correlationId set in the request by the user
+     * @param request
+     */
+    static validateUserSetServerParams(scopes: Array<string>, state: string, correlationId: string, clientId: string): UserSetRequestParams {
```

Comment:
A new type feels a bit heavy-handed here. An alternative would be to have these parameters passed in as before, and then call this function from the constructor to validate?

Reply:
It is getting complicated with the dependencies with `getCachedTokenInternal`. I implemented that and rewrote the whole code.